src/main/java/bio/terra/service/resourcemanagement/azure/AzureAuthService.java

```diff
@@ -43,7 +43,7 @@ public AzureAuthService(AzureResourceConfiguration configuration) {
         new RequestRetryOptions(
             RetryPolicyType.EXPONENTIAL, maxRetries, retryTimeoutSeconds, null, null, null);
     // wrap the cache map with a synchronized map to safely share the cache across threads
-    authorizedMap = Collections.synchronizedMap(new PassiveExpiringMap<>(24, TimeUnit.HOURS));
+    authorizedMap = Collections.synchronizedMap(new PassiveExpiringMap<>(15, TimeUnit.MINUTES));
```

Comment:
That's a big change! Was the 24 hour timeout a placeholder that we were planning to change all along?

Reply:
It was initially an arbitrary number and it led to some bad behavior when deleting and recreating a storage account with the same name (the key would be new and there's no way to notify the cache that it's no longer valid...this seemed like the lightest change.  It also turns out that the per impact of getting the key isn't super significant)